swift/Sources/FlatBuffers/ByteBuffer.swift

```diff
@@ -175,12 +175,23 @@ public struct ByteBuffer {
     /// - Parameters:
     ///   - value: Pointer to the object in memory
     ///   - size: Size of Value being written to the buffer
+    @available(*, deprecated, message: "0.9.0 will be removing the following method. Regenerate the code")
     @usableFromInline mutating func push(struct value: UnsafeMutableRawPointer, size: Int) {
         ensureSpace(size: size)
         memcpy(_storage.memory.advanced(by: writerIndex &- size), value, size)
         defer { value.deallocate() }
         _writerSize = _writerSize &+ size
     }
+    
+    @usableFromInline mutating func prepareBufferToReceiveStruct(of size: Int) {
+        ensureSpace(size: size)
```

Comment:
How does this align correctly? For example, `size` could be 16, but the alignment might need to be to 8, depending on the elements of the struct.

Also, does this guarantee that parts of the struct not written are 0, since you're only writing the actual fields? For example if you serialize a vtable, then discard it, then write a struct, you may end up with non-0 bytes?

If these two things are messy to solve, you may consider implementing it in the same way e.g. Java does, by pre-aligning the struct at the start, and then writing the elements in reverse (without offsets) and with explicit padding.

Reply:
Regarding the alignment, it's being taken care of by the following method:
```swift
    mutating public func createStructOf(size: Int, alignment: Int) {
         preAlign(len: size, alignment: alignment)
         _bb.prepareBufferToReceiveStruct(of: size)
     }
```
since when I was building the library earlier this year I faced an issue when we are writing the data in the normal flow that FlatBuffers write to the buffer (back to front). so this will basically make the write pointer point into the current write index + the struct size. then we will write the data into that space going forward instead of backwards.

```
                    | size of struct to be written
current_write_index <- 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 <- last_written_byte
                    -> we will be writing forward instead
```

and yes it's guaranteed that the parts that aren't written are zero since we actually initialize the memory with zeros before writing to the buffer. 
Note: I will need to look into the discarding part

Reply:
```swift
  func endTable() {
  ...
        if let offset = isAlreadyAdded {
            let vTableOff = Int(vTableOffset)
            let space = _bb.capacity &- vTableOff
            _bb.write(value: Int32(offset &- vTableOff), index: space, direct: true)
            _bb.resize(_bb.capacity &- space)
        } else {
            _bb.write(value: Int32(vt_use &- vTableOffset), index: Int(vTableOffset))
            _vtables.append(_bb.size)
        }
  ...
  }
```

```swift
struct ByteBuffer {
    /// Resizes the buffer size
    /// - Parameter size: new size for the buffer
    @usableFromInline mutating internal func resize(_ size: Int) {
        assert((_writerSize &- size) > 0, "New size should NOT be a negative number")
        var zero: UInt8 = 0
        for i in 0..<(_writerSize &- size) {
            memcpy(_storage.memory.advanced(by: writerIndex &+ i), &zero, MemoryLayout<UInt8>.size)
        }
        _writerSize = size
    }
}
```
So endTable is the only place where this would happen 

>  if you serialize a vtable, then discard it, then write a struct

and since we already resize and overwrite the bytes we wrote earlier we would not have the issue of having non zero-bytes

Reply:
do you also overwrite any padding/alignment bytes?

Reply:
Yes, since the resize function will take a size that should be overwrite on top of it. since what we pass to it is the following `let space = _bb.capacity &- vTableOff; _bb.resize(_bb.capacity &- space)`

Reply:
Hmm that's weird to have a generic sounding method called `resize` that deals with clearing unused vtable memory? And is repeatedly calling `memcpy` on a single byte the best way to clear memory in Swift?

But either way, it looks like you've taken care of these cases, so that is good enough for me.

Reply:
I've renamed the function and also used `memset` instead of `memcpy` that definitely makes more sense.